test/uk/gov/hmrc/transitmovementsauditing/models/AuditTypeSpec.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsauditing.models
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+
+class AuditTypeSpec extends AnyFreeSpec with Matchers {
+
+  "every AuditType must match the object name" - AuditType.values.foreach {
```

Comment:
Looks good to me, just one question:
Do we use .toString anywhere? Just wondering if these tests are useful?

Reply:
yes with API test cases Dan has written same way.

Reply:
@drnaylor-oc how come we verify that toString matches the object name? I'm sure there's a good reason but I just want to understand it :)

Reply:
the comment explains it:

`// toString prints the case object's identifier, so this is what we want`

If you have a `case object`, the name of the case object is also the output of `toString`. I was verifying that the audit type name matched the name of case object, there was a case where it wasn't which is why the test was introduced. Of course, the alternative would have just been to use `toString` where we want to get the audit type out, but we'd gone down this road before I realised we could do that.

Basically, if we have

```scala
case object AnObject
```

then `AnObject.toString` is `"AnObject"`.